examples/markdown_challenges.md

```diff
@@ -52,6 +52,125 @@ an HTML href tag that uses `http_base`.
 - Organization: ACI
 - Created: 2020-06-24
 
+## Network Options
+
+This optional section can be used to configure the Docker network associated
+with each instance of this challenge. The available options are listed below,
+along with example usage.
+
+The `Internal` option disables Internet access, restricting network traffic to
+the instance's containers. This is equivalent to passing the [`--internal`](https://docs.docker.com/engine/reference/commandline/network_create/#network-internal-mode) flag to [`docker network create`](https://docs.docker.com/engine/reference/commandline/network_create/). Specify a boolean value, as shown below. Defaults to `false`.
+
+- Internal: true
+
+## Container Options
```

Comment:
What are your thoughts on just handing off the contents of this section to a YAML parser?  It should still look/feel similar to the rest of the Markdown file but could greatly simplify reading all of the information in and provide a clear continuity mechanism with its representation in the JSON format.  My thought would it would look something like

```yaml
## Container Options

init: true
cpus: 0.5
memory: 128m
ulimits:
    nofile: 512
    stack: 4096
overrides:
    web:
        cpus: 1
    db:
        memory: 512m
```

If you're good with this, I can take on updating what you already have to go down this path.

Reply:
This seems like a good idea to me - I'm also not a big fan of all the handrolled psuedo-YAML parsing I added to `loader_markdown` for this :)

One suggestion would be that maybe a yaml code fence block should be required though, to make it clear that we're entering a separate language/parsing context and to make the rendered markdown cleaner:

````markdown
## Container Options

```yaml
init: true
cpus: 0.5
memory: 128m
```

````

Reply:
That seems fair.  Do you want me to work on reworking what you have to this setup?

Reply:
Sure. I'm working on the `StorageOpts` -> `NoNewPrivileges` + environment variable tweak right now but should have that finished and pushed up shortly.

Reply:
I pushed the update to change over to parsing from YAML.  Some changes from current behavior:

1. `Container Options` -> `Challenge Options` (since it includes network options)
2. `internal` -> `internal_network` (since it shows up next to container options and is slightly ambiguous in that context)

On a different note, I don't think the internal networking option is as useful as we may have first thought.  It silently kills exposing ports and fully isolates the network even when exposed ports were explicitly requested.

Reply:
Sigh - you're right, the internal network option does seem pretty useless after all. Seems like there's no real reason to keep the whole networkOptions table then, what do you think? I think I'll go back to seeing if there's an iptables approach that I could use for limiting outbound access.

Reply:
I think the YAML change looks good and simplifies things. I do want to tweak the option documentation a bit now though, and I think `ContainerOptionsWrapper` can just become a regular `map[string]ContainerOptions` since it no longer needs to handle the non-map case (since it's always the override map if present)

Reply:
That sounds good.  I'm inclined to leave the `NetworkOptions` struct as an empty placeholder, though.  That way if we decide in the future to support IPv6 or another network knob becomes available the API side of it should be pretty clear.

Reply:
Removed `internal_network` option and associated database table (but left stub struct and commented-out blocks for saving/loading from DB) in https://github.com/ArmyCyberInstitute/cmgr/pull/33/commits/c50cbdd3939677c63e2a4a0b4b0755de95859146